### DISCUSSION TITLE
Node.js Error: Undefined Symbol when Loading Module

### DISCUSSION THREAD

**Discussion by keerthanakumar-08** [directus/directus#21813] [Archive]
### Summary

Hello Directus Community,

I've been using Directus version 10.9.2 for some time now, and everything was working smoothly until recently. However, I've encountered an issue where my existing code is no longer functioning as expected.

### Basic Example

![Screenshot from 2024-03-12 09-40-11](https://github.com/directus/directus/assets/98075297/e8fe2289-e70e-463e-943e-6ffc48281632)

### Motivation

Trying to update new version of directus  same issue happened.

### Detailed Design

Every command of directus facing same issue my existing code not working properly

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

![Screenshot from 2024-03-12 09-44-57](https://github.com/directus/directus/assets/98075297/4345b1d6-26ad-4110-a817-fbc9222c869d)

### Alternatives

![image](https://github.com/directus/directus/assets/98075297/a6f08069-a1af-4b23-8ed3-6645d170a31d)

### Adoption Strategy

It's essential to carefully consider the implications of the proposal on existing Directus developers and projects. Providing clear guidance and support during the adoption process will help facilitate a smooth transition and minimize any potential disruptions to the ecosystem.
Encourage collaboration and feedback from the Directus community to gather insights and address any concerns or challenges that may arise during the adoption phase. Open communication channels and forums for discussion can facilitate dialogue and foster a sense of community involvement in the adoption proces

**Comment by br41nslug** [comment]
The error seems to indicate that you may be using an unsupported version NodeJS for the `isolated-vm` dependency https://github.com/laverdet/isolated-vm?tab=readme-ov-file#requirements.

Closing this as it is not a feature request.

**Comment by keerthanakumar-08** [comment]
![image](https://github.com/directus/directus/assets/98075297/0d8c6a9a-1082-4d54-80cf-170f94a172e2)
But I'm using node version 16.
![image](https://github.com/directus/directus/assets/98075297/0bef32c0-f31e-4929-ab0b-db676517da7d)

**Reply by paescuj** [reply]
Directus currently requires Node.js 18, see https://docs.directus.io/self-hosted/cli.html#requirements
